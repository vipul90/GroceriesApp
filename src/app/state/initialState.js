import GroceryDetailItem from '../Helpers/GroceryDetailItem'
import DateValue from '../Helpers/Utils'
import Paneer from '../grocery/assests/Paneer.jpg';

const initialState = {
    loading: false,
    error: null,
    availableGroceryList:[
        new GroceryDetailItem(1,'Paneer',2,DateValue,80.50,'200 gm',Paneer),
        new GroceryDetailItem(2,'Item 2',3,DateValue,200,'2 Kg'),
        new GroceryDetailItem(3,'Item 3',4,DateValue,300,'1 l'),
        new GroceryDetailItem(4,'Item 4',5,DateValue,400,'2 l'),
        new GroceryDetailItem(5,'Item 5',5,DateValue,400,'2 l'),
        new GroceryDetailItem(6,'Item 6',30,DateValue,1400,'5 Kg'),
        new GroceryDetailItem(7,'Item 7',40,DateValue,1400,'5 Kg'),
        new GroceryDetailItem(8,'Item 8',10,DateValue,1400,'5 Kg'),
        new GroceryDetailItem(9,'Item 9',13,DateValue,1400,'5 Kg'),
        new GroceryDetailItem(10,'Item 10',12,DateValue,1400,'5 Kg'),
        new GroceryDetailItem(11,'Item 11',20,DateValue,1400,'5 Kg'),
        new GroceryDetailItem(12,'Item 12',21,DateValue,1400,'5 Kg'),
        new GroceryDetailItem(13,'Item 13',23,DateValue,1400,'5 Kg'),
        new GroceryDetailItem(14,'Item 14',5,DateValue,1400,'5 Kg'),
        new GroceryDetailItem(15,'Item 15',20,DateValue,1400,'5 Kg'),
        new GroceryDetailItem(16,'Item 16',5,DateValue,1400,'5 Kg'),
        new GroceryDetailItem(17,'Item 17',60,DateValue,1400,'5 Kg'),
        new GroceryDetailItem(18,'Item 18',19,DateValue,1400,'5 Kg'),
        new GroceryDetailItem(19,'Item 19',20,DateValue,1400,'5 Kg'),
        new GroceryDetailItem(20,'Item 20',40,DateValue,1400,'5 Kg'),
        new GroceryDetailItem(21,'Item 21',16,DateValue,1400,'5 Kg'),
        new GroceryDetailItem(22,'Item 22',100,DateValue,1400,'5 Kg'),
        new GroceryDetailItem(23,'Item 23',400,DateValue,1400,'5 Kg'),
        new GroceryDetailItem(24,'Item 24',220,DateValue,1400,'5 Kg'),

    ],
    userGroceryItems: [
        // new GroceryDetailItem(1,'Item 1',2,DateValue),
        // new GroceryDetailItem(2,'Item 2',3,DateValue),
        // new GroceryDetailItem(3,'Item 3',4,DateValue)
    ],
    CurrentItemId:0
  };

  export default initialState;