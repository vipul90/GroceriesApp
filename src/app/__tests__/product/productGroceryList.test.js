// import React from 'react';
// import { shallow } from 'enzyme';
// import { findByTestAtrr } from '../../services/testUtil';
// import ProductGroceryList from '../../grocery/components/product/ProductGroceryList';
// import initialState from '../../state/initialState'

// const componentSetUp = () => {
//   const props = {
//     values: { email: '', password: '' },
//     handleChange: jest.fn(),
//     handleSubmit: jest.fn(),
//     handleBlur: jest.fn(),
//     errors: { email: '', password: '' },
//     touched: { email: '', password: '' },
//     loggingIn: false
//   };
//   const component = shallow(<ProductGroceryList {...props} />);
//   return component;
// };

// describe('Product Grocery List Component', () => {
//   let componentInstance;
//   beforeEach(() => {
//     componentInstance = componentSetUp();
//   });

//   it('Should render without errors', () => {
//     const wrapper = findByTestAtrr(componentInstance, 'ProductGroceryListComponent');
//     expect(wrapper.length).toBe(1);
//   });
// });

import React from 'react';
import { Provider } from 'react-redux';
// import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import ProductGroceryList from '../../grocery/components/product/ProductGroceryList';//ProductGroceryList
import initialState from '../../state/initialState'
// import MyConnectedComponent from '.';
import toJson from 'enzyme-to-json'; //added this line
// import { createRender } from '@material-ui/core/test-utils';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});

const mockStore = configureStore([]);
 
describe('My Connected React-Redux Component', () => {
  let store;
  let component;
 
  beforeEach(() => {
    store = mockStore({
        GroceryDetails : initialState
    });
    // const render = createRender();

    component = shallow(
      <Provider store={store}>
        <ProductGroceryList />
      </Provider>
    );
  });
 
  it('should render with given state from Redux store', () => {
        // expect(component.toJSON()).toMatchSnapshot();
        expect(toJson(component)).toMatchSnapshot();
  });
 
//   it('should dispatch an action on button click', () => {
 
//   });
});
