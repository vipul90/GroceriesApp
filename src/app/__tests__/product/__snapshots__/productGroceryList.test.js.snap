// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(ProductGroceryList) />
</ContextProvider>
`;
